src/ucp/core/ucp_worker.c

```diff
@@ -1890,8 +1890,6 @@ ucs_status_t ucp_worker_wait(ucp_worker_h worker)
     UCP_CONTEXT_CHECK_FEATURE_FLAGS(worker->context, UCP_FEATURE_WAKEUP,
                                     return UCS_ERR_INVALID_PARAM);
 
-    UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
```

Comment:
the code below accesses worker and context fields, it should be with lock

Reply:
But `ucp_worker_arm` has locking on worker: https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_worker.c#L1847

Maybe move that lock before read?

Reply:
what about lines 1903-1908?